Makefile

```diff
@@ -41,8 +41,10 @@ $(shell perl -pi -e 's/memory:.*/memory: $(KSERVE_CONTROLLER_MEMORY_LIMIT)/' con
 all: test manager agent router
 
 # Run tests
+# TODO: We can remove the `arch` flag once we upgraded $ENVTEST_K8S_VERSION to 1.24 or higher.
```

Comment:
@tenzen-y shall we just upgrade the envtest k8s version here in the makefile? This is only for local testing so there should be no risk to update it.

Reply:
@yuzisun We are using `make test` in CI. So, we might be better to keep using K8s 1.23 in Makefile.
wdyt?

https://github.com/kserve/kserve/blob/7c199b9996fc900a852a74dc806f0610142ffcf4/.github/workflows/go.yml#L39-L41

Reply:
@yuzisun Thanks for clarifying! Sounds good.
I will remove the `--arch` flag and then upgrade the K8s version for the envtest and minikube to 1.24.